crates/flux-fhir-analysis/src/annot_check.rs

```diff
@@ -283,61 +304,79 @@ impl<'zip> Zipper<'zip> {
 
     fn zip_bty(
         &mut self,
+        genv: &GlobalEnv,
         bty: &fhir::BaseTy,
         expected_bty: &'zip fhir::BaseTy,
     ) -> Result<(), ErrorGuaranteed> {
         match (&bty.kind, &expected_bty.kind) {
             (fhir::BaseTyKind::Path(qpath), fhir::BaseTyKind::Path(expected_qpath)) => {
-                self.zip_qpath(qpath, expected_qpath)
+                self.zip_qpath(genv, qpath, expected_qpath)
             }
             (fhir::BaseTyKind::Slice(ty), fhir::BaseTyKind::Slice(expected_ty)) => {
-                self.zip_ty(ty, expected_ty)
+                self.zip_ty(genv, ty, expected_ty)
             }
             _ => Err(self.emit_err(errors::InvalidRefinement::from_btys(bty, expected_bty))),
         }
     }
 
     fn zip_qpath(
         &mut self,
+        genv: &GlobalEnv,
         qpath: &fhir::QPath,
         expected_qpath: &'zip fhir::QPath,
     ) -> Result<(), ErrorGuaranteed> {
         match (qpath, expected_qpath) {
             (fhir::QPath::Resolved(None, path), fhir::QPath::Resolved(None, expected_path)) => {
-                self.zip_path(path, expected_path)
+                self.zip_path(genv, path, expected_path)
             }
             (
                 fhir::QPath::Resolved(Some(self_ty), path),
                 fhir::QPath::Resolved(Some(expected_self_ty), expected_path),
             ) => {
-                self.zip_ty(self_ty, expected_self_ty)?;
-                self.zip_path(path, expected_path)
+                self.zip_ty(genv, self_ty, expected_self_ty)?;
+                self.zip_path(genv, path, expected_path)
             }
             _ => Err(self.emit_err(errors::InvalidRefinement::from_qpaths(qpath, expected_qpath))),
         }
     }
 
+    fn is_same_res(genv: &GlobalEnv, res: Res, expected: Res) -> bool {
```

Comment:
Ok, this is a bit dumb. The only reason we need to check this is that we allow you to write the "return type" for a variant. If we instead had:

```rust
#[flux::refined_by(b: bool)]
enum Option<T> {
    #[flux::variant([false])]
    None,
    #[flux::variant((T) -> [true])]
    Some(T)
}
```
then, this wouldn't cause any problems. Not in this PR, but we should consider changing the syntax.